test/token/FTCVesting.spec.ts

```diff
@@ -0,0 +1,160 @@
+import "module-alias/register";
+import { BigNumber } from "@ethersproject/bignumber";
+
+import { Account } from "@utils/types";
+import { IndexToken, FTCVesting } from "@utils/contracts";
+import DeployHelper from "@utils/deploys";
+import {
+  addSnapshotBeforeRestoreAfterEach,
+  ether,
+  getLastBlockTimestamp,
+  getRandomAccount,
+  getWaffleExpect,
+} from "@utils/index";
+import { ContractTransaction } from "ethers";
+
+const expect = getWaffleExpect();
+
+describe("FTCVesting", () => {
+  let owner: Account;
+  let treasury: Account;
+  let recipient: Account;
+
+  let deployer: DeployHelper;
+  let index: IndexToken;
+
+  before(async () => {
+    owner = await getRandomAccount();
+    treasury = await getRandomAccount();
+    recipient = await getRandomAccount();
+
+    deployer = new DeployHelper(owner.wallet);
+
+    index = await deployer.token.deployIndexToken(owner.address);
+  });
+
+  addSnapshotBeforeRestoreAfterEach();
+
+  describe("#constructor", async () => {
+    let subjectVestingAmount: BigNumber;
+    let subjectVestingStart: BigNumber;
+    let subjectVestingCliff: BigNumber;
+    let subjectVestingEnd: BigNumber;
+
+    beforeEach(async () => {
+      const now = await getLastBlockTimestamp();
+      subjectVestingStart = now;
+      subjectVestingCliff = now.add(60 * 60 * 24 * 183);
+      subjectVestingEnd = now.add(60 * 60 * 24 * 547);
+      subjectVestingAmount = ether(100);
+    });
+
+    async function subject(): Promise<FTCVesting> {
+      return await deployer.token.deployFtcVesting(
+        index.address,
+        recipient.address,
+        treasury.address,
+        subjectVestingAmount,
+        subjectVestingStart,
+        subjectVestingCliff,
+        subjectVestingEnd,
+      );
+    }
+
+    it("should set the correct state variables", async () => {
+      const ftcVest = await subject();
+
+      expect(await ftcVest.recipient()).to.eq(recipient.address);
+      expect(await ftcVest.treasury()).to.eq(treasury.address);
+      expect(await ftcVest.vestingAmount()).to.eq(subjectVestingAmount);
+      expect(await ftcVest.vestingBegin()).to.eq(subjectVestingStart);
+      expect(await ftcVest.vestingEnd()).to.eq(subjectVestingEnd);
+      expect(await ftcVest.vestingCliff()).to.eq(subjectVestingCliff);
+      expect(await ftcVest.lastUpdate()).to.eq(subjectVestingStart);
+    });
+  });
+
+  xdescribe("#setRecipient", async () => {
```

Comment:
This type will break your tests

Reply:
you mean the x? that was intentional (for now) to test things and single out the #claim test

Reply:
Oh I see, I did not realize that it was intentional